File: src/examples/ruby/shipments/buy_shipment.rb
# frozen_string_literal: true

require 'easypost'
require 'dotenv'
require 'amazing_print'

Dotenv.load('/Users/jhammond/git/easypost/easypost-tools/.env')
EasyPost.api_key = ENV['EASYPOST_TEST_API_KEY']

shipment = EasyPost::Shipment.retrieve('shp_123...')
shipment.buy(
  rate: shipment.lowest_rate
)

ap shipment

File: src/examples/php/shipments/buy-shipment.php
<?php

require_once '/Users/jhammond/git/easypost/easypost-tools/vendor/autoload.php';

use Dotenv\Dotenv;
use EasyPost\EasyPost;
use EasyPost\Error;
use EasyPost\Shipment;

$dotenv = Dotenv::createImmutable('/Users/jhammond/git/easypost/easypost-tools');
$dotenv->load();
EasyPost::setApiKey($_ENV['EASYPOST_TEST_API_KEY']);

try {
    $shipment = \EasyPost\Shipment::retrieve("shp_123...");
    $shipment->buy([
        'rate'      => $shipment->lowest_rate(), // alternatively, provide a rate ID here instead
        //   'insurance' => 249.99 // optional insurance
    ]);
    echo $shipment;
} catch (Error $exception) {
    echo $exception;
}
